src/main/java/com/unb/tracker/web/CourseController.java

```diff
@@ -84,12 +95,69 @@ Course getCourse(@PathVariable Long courseId) {
     @PostMapping(value = "/courses/{courseId}/seats")
     public @ResponseBody
     Course postCourseSeats(@PathVariable Long courseId, @RequestBody List<Seat> seats) {
+        LOG.info("postCourseSeats - starting");
         Course course = courseRepository.findOne(courseId);
+
+        for(Seat s : seats) {
+            s.setCourse(course);
+        }
+        seatRepository.save(seats);
+
         course.setSeats(seats);
         courseRepository.save(course);
+
         return course;
     }
 
+    @PostMapping(value = "/courses/{courseId}/seat")
+    public @ResponseBody
+    String postCourseSeat(@PathVariable Long courseId, @RequestBody Seat seat, Principal principal) {
+        LOG.info("postCourseSeats - starting - seat.id: {}", seat.getId());
+
+        User user = userRepository.findByUsername(principal.getName());
+        if(seat.getStudent() != null && !user.getId().equals(seat.getStudent().getId())) {
+            LOG.warn("{} trying to alter {}'s seat", user.getUsername(), seat.getStudent().getUsername());
+            throw new BadRequestException();
+        }
+
+        Course course = courseRepository.findOne(courseId);
+        List<Seat> seats = course.getSeats();
+
+        // Are we trying to remove a student?
+        if(seat.getStudent() == null) {
+            for(Seat s : seats) {
+                if(s.getId().equals(seat.getId())) {
+                    LOG.debug("Found the matching seat! Now confirming the logged in user owns that seat");
+                    if(s.getStudent().getId().equals(user.getId())) {
```

Comment:
This logic get's pretty wacky. I understand what is happening, but you could refactor these nested conditional blocks with guard clauses. It would help make it a bit more readable :)

Reply:
Sounds good. I will refactor this now